src/reducers/sources.js

```diff
@@ -322,20 +354,22 @@ export function hasPrettySource(state: OuterState, id: string) {
 export function getSourceByUrlInSources(
   sources: SourcesMap,
   urls: UrlsMap,
-  url: string
+  url: string,
+  isOriginal: boolean
 ) {
   const foundSources = getSourcesByUrlInSources(sources, urls, url);
   if (!foundSources) {
     return null;
   }
 
-  return foundSources[0];
+  return foundSources.find(source => isOriginalId(source.id) == isOriginal);
```

Comment:
we can switch this case to `isOriginal` as well, which will prevent `:formatted` edge cases